From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Niklas <57962465+niklasnieberler@users.noreply.github.com>
Date: Thu, 22 Aug 2024 22:18:44 +0200
Subject: [PATCH] remove help command


diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index 13fb3e3a329910c41adb6ddaafbca3ccba6bfe22..fa9359acb4186fc2eb5bb7cf87a3e5b784cb2209 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -43,7 +43,7 @@ public class SimpleCommandMap implements CommandMap {
     }
 
     public void setFallbackCommands() {
-        register("bukkit", new HelpCommand());
+       // register("bukkit", new HelpCommand()); CocoPaper
     }
 
     /**
