From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: niklasnieberler <nikigalaxylp@gmail.com>
Date: Thu, 22 Aug 2024 10:21:48 +0200
Subject: [PATCH] add isMoving and getLastMoving in Player


diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 6c280abdef5f80b668d6090f9d35283a33e21e0c..64c0cf384f39760975a7a4661e1f8f8f2020e3ee 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -283,6 +283,11 @@ public class ServerPlayer extends net.minecraft.world.entity.player.Player imple
     // Paper end - Optional per player mob spawns
     public final int[] mobBackoffCounts = new int[MOBCATEGORY_TOTAL_ENUMS]; // Paper - per player mob count backoff
 
+    // forktest start
+    public Boolean isMoving = false;
+    public Long lastMoving = 0L;
+    // forktest end
+
     // CraftBukkit start
     public CraftPlayer.TransferCookieConnection transferCookieConnection;
     public String displayName;
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index b13057c0792067cc6b0abdf0d64a9be2cc9389a4..e315ca045454bbd10444f0b9ce8554892680a0ba 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -662,6 +662,9 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                     PlayerMoveEvent event = new PlayerMoveEvent(player, from, to);
                     this.cserver.getPluginManager().callEvent(event);
 
+                    this.player.isMoving = (fromX != toX || fromY != toY || fromZ != toZ);
+                    this.player.lastMoving = System.currentTimeMillis();
+
                     // If the event is cancelled we move the player back to their old location.
                     if (event.isCancelled()) {
                         this.teleport(from);
@@ -1560,6 +1563,9 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                                     PlayerMoveEvent event = new PlayerMoveEvent(player, from, to);
                                     this.cserver.getPluginManager().callEvent(event);
 
+                                    this.player.isMoving = (d0 != toX || d1 != toY || d2 != toZ);
+                                    this.player.lastMoving = System.currentTimeMillis();
+
                                     // If the event is cancelled we move the player back to their old location.
                                     if (event.isCancelled()) {
                                         this.teleport(from);
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 99dbfa2dd5e1e41d7d06bd86fc9c11bbde3d9558..5c7c00919ad3b0bef0b6ff4cf8319c5b792e9182 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -3096,11 +3096,22 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
     // Paper end
 
-    // forktes tstart
+    // forktest start
     @Override
-    public PlayerData getPlayerData() {
+    public @NotNull PlayerData getPlayerData() {
         return this.playerData;
     }
+
+    @Override
+    public boolean isMoving() {
+        return getHandle().isMoving;
+    }
+
+    @Override
+    public @NotNull Long getLastMoving() {
+        return getHandle().lastMoving;
+    }
+
     // forktest end
 
     // Paper start
