From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: niklasnieberler <nikigalaxylp@gmail.com>
Date: Thu, 22 Aug 2024 10:47:22 +0200
Subject: [PATCH] add PlayerTickEvent and ServerTickEvent


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index bd7b0b7eb3ef458acb9dbb52415dc871165a75aa..aa10c3f07b394749bb736421c620db14fa889156 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -35,6 +35,9 @@ import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 import java.util.UUID;
+import java.util.concurrent.Executors; // forktest
+import java.util.concurrent.ScheduledExecutorService; // forktest
+import java.util.concurrent.TimeUnit; // forktest
 import java.util.function.Consumer;
 import java.util.logging.Level;
 import java.util.logging.Logger;
@@ -262,6 +265,8 @@ import org.yaml.snakeyaml.error.MarkedYAMLException;
 
 import net.md_5.bungee.api.chat.BaseComponent; // Spigot
 import world.estaria.paper.DependencyManager; // forktest
+import world.estaria.paper.event.player.PlayerTickEvent;
+import world.estaria.paper.event.server.ServerTickEvent;
 
 import javax.annotation.Nullable; // Paper
 import javax.annotation.Nonnull; // Paper
@@ -417,6 +422,17 @@ public final class CraftServer implements Server {
         this.pluginManager.paperPluginManager = this.paperPluginManager;
          // Paper end
 
+        // forktest start
+        final ScheduledExecutorService schedulerService = Executors.newScheduledThreadPool(1);
+        schedulerService.scheduleAtFixedRate(() -> {
+            new ServerTickEvent().callEvent();
+
+            for (Player player : Bukkit.getOnlinePlayers()) {
+                new PlayerTickEvent(player).callEvent();
+            }
+        }, 0, 1, TimeUnit.MILLISECONDS);
+        // forktest end
+
         CraftRegistry.setMinecraftRegistry(console.registryAccess());
 
         if (!Main.useConsole) {
